src/elements/content-sidebar/MetadataSidebarRedesign.tsx

```diff
@@ -76,18 +76,38 @@ function MetadataSidebarRedesign({
     const [isUnsavedChangesModalOpen, setIsUnsavedChangesModalOpen] = React.useState<boolean>(false);
     const [selectedTemplates, setSelectedTemplates] =
         React.useState<Array<MetadataTemplateInstance | MetadataTemplate>>(templateInstances);
+    const [pendingTemplateToEdit, setPendingTemplateToEdit] = React.useState<MetadataTemplateInstance | null>(null);
 
     React.useEffect(() => {
         setSelectedTemplates(templateInstances);
     }, [templateInstances]);
 
-    const handleUnsavedChanges = () => {
-        setIsUnsavedChangesModalOpen(true);
+    const handleTemplateSelect = (selectedTemplate: MetadataTemplate) => {
+        if (editingTemplate) {
+            setPendingTemplateToEdit(selectedTemplate as MetadataTemplateInstance);
+            setIsUnsavedChangesModalOpen(true);
+        } else {
+            setSelectedTemplates([...selectedTemplates, selectedTemplate]);
+            setEditingTemplate(selectedTemplate as MetadataTemplateInstance);
```

Comment:
For future, as soon as @karolinaru merges https://github.com/box/box-ui-elements/pull/3663/ we should use `convertTemplateToTemplateInstance` here.